src/WhiteLibrary/__init__.py

```diff
@@ -169,8 +169,34 @@ def __init__(self, screenshot_dir=None):
                           UiItemKeywords(self),
                           WindowKeywords(self),
                           ScreenshotKeywords(self, screenshot_dir)]
+        self._running_keyword = None
+        self._running_on_failure_keyword = False
         DynamicCore.__init__(self, self.libraries)
 
+    def run_keyword(self, name, args, kwargs):  # pylint: disable=signature-differs
+        """Reimplemtation of run_keyword.
+
+        calls robot framework's own implementation but handles screenshots if/when exceptions are triggered.
+        """
+        self._running_keyword = name
+        try:
+            return DynamicCore.run_keyword(self, name, args, kwargs)
+        except Exception:
+            self._failure_occurred()
+            raise
+        finally:
+            self._running_keyword = None
+
+    def _failure_occurred(self):
+        if self._running_on_failure_keyword:
+            return
+        try:
+            self._running_on_failure_keyword = True
```

Comment:
Maybe there could be a comment on why this is used, to avoid possible future confusion :)

Reply:
will add!